amulet/api/player/player_manager.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+from typing import Tuple, Generator
+
+import weakref
+
+from amulet.api.history import Changeable
+from amulet.api.history.data_types import EntryKeyType, EntryType
+from amulet.api.history.history_manager import DatabaseHistoryManager
+from amulet.api import level as api_level
+
+LOCAL_PLAYER = "~local_player"
+
+
+class Player(Changeable):
+    def __init__(
+        self,
+        _uuid: str,
+        position: Tuple[float, float, float],
+        rotation: Tuple[float, float],
+    ):
+        """
+        Creates a new instance of :class:`Player` with the given UUID, position, and rotation
+
+        :param _uuid: The UUID of the player
+        :param position: The position of the player in world coordinates
+        :param rotation: The rotation of the player
+        """
+        super().__init__()
+        self._uuid = _uuid
+        self._position = position
+        self._rotation = rotation
+
+    @property
+    def uuid(self) -> str:
+        """The player's UUID"""
+        return self._uuid
+
+    @property
+    def position(self) -> Tuple[float, float, float]:
+        """The current position of the player in the world"""
+        return self._position
+
+    @property
+    def rotation(self) -> Tuple[float, float]:
+        """The current rotation of the player in the world"""
+        return self._rotation
+
+
+class PlayerManager(DatabaseHistoryManager):
+    def __init__(self, level: api_level.BaseLevel):
+        """
+        Construct a new :class:`PlayerManager` instance
+
+        This should not be used by third party code
+
+        :param level: The world that this player manager is associated with
+        """
+        super().__init__()
+        self._level = weakref.ref(level)
+
+    @property
+    def level(self) -> api_level.BaseLevel:
+        """The level that this player manager is associated with."""
+        return self._level()
+
+    def get_players(self) -> Generator[str, None, None]:
+        """
+        Returns a generator of all player ids that are present in the level
+        """
+        yield from self.level.level_wrapper.get_players()
```

Comment:
I have just spotted that this is side-stepping the loaded and cached versions which might cause some issues.
Looking into it I will probably need to do a little bit of rewriting of the chunk manager class so that these two can share some of the logic.

For future reference this affects has_chunk and all_chunk_coords,